File: scala/sdktest/OutputActivatePortalModuleEntityTest.scala
// Generated basic-flow test for the output_activate_portal_module entity (model-driven;
// mirrors the java TestEntity generator). A dependency-free scala-cli test
// object driven by SdkEntityTestMain. Runs against the in-memory test
// transport seeded with the shipped OutputActivatePortalModuleTestData.json fixtures.

import java.util.{ArrayList, LinkedHashMap, List => JList, Map => JMap}

import voxgig.bluefintecsuserbackofficesdk.core.{Helpers, BluefinTecsUserBackofficeSDK}
import voxgig.bluefintecsuserbackofficesdk.utility.struct.Struct

object OutputActivatePortalModuleEntityTest {

  def run(rep: SdkTestReport): Unit = {
    rep.scope("output_activate_portal_module.instance") {
      val testsdk = BluefinTecsUserBackofficeSDK.testSDK()
      val ent = testsdk.outputActivatePortalModule(null)
      rep.check("output_activate_portal_module.instance", ent != null, "expected non-null output_activate_portal_module entity")
    }

    rep.scope("output_activate_portal_module.basic") {
      val entityData = Helpers.toMapAny(SdkTestSupport.readJson(
          "../.sdk/test/entity/output_activate_portal_module/OutputActivatePortalModuleTestData.json"))
      val options = new LinkedHashMap[String, Object]()
      options.put("entity", entityData.get("existing"))
      val client = BluefinTecsUserBackofficeSDK.testSDK(options, null)

      val idmap = new LinkedHashMap[String, Object]()
      idmap.put("output_activate_portal_module01", "OUTPUT_ACTIVATE_PORTAL_MODULE01")
      idmap.put("output_activate_portal_module02", "OUTPUT_ACTIVATE_PORTAL_MODULE02")
      idmap.put("output_activate_portal_module03", "OUTPUT_ACTIVATE_PORTAL_MODULE03")
      val now = System.currentTimeMillis()

      // CREATE
      val outputActivatePortalModuleRef01Ent = client.outputActivatePortalModule(null)
      var outputActivatePortalModuleRef01Data = Helpers.toMapAny(Struct.getprop(
          Struct.getpath(entityData, "new.output_activate_portal_module"), "output_activate_portal_module_ref01"))
      val outputActivatePortalModuleRef01DataResult = outputActivatePortalModuleRef01Ent.create(outputActivatePortalModuleRef01Data, null)
      outputActivatePortalModuleRef01Data = Helpers.toMapAny(outputActivatePortalModuleRef01DataResult)
      rep.check("output_activate_portal_module.create.map", outputActivatePortalModuleRef01Data != null, "expected create result to be a map")
    }
  }
}

File: dart/test/entity/output_activate_portal_module/OutputActivatePortalModuleEntity_test.dart
// ignore_for_file: unused_import, unused_local_variable, non_constant_identifier_names

import 'dart:convert';
import 'dart:io';

import '../../harness.dart';
import '../../utility.dart';

import '../../../lib/BluefinTecsUserBackofficeSDK.dart';
import '../../../lib/utility/voxgig_struct.dart' as vs;

void tests() {
  describe('OutputActivatePortalModuleEntity', () {
    test('instance', (t) async {
      final testsdk = BluefinTecsUserBackofficeSDK.test();
      final ent = testsdk.OutputActivatePortalModule();
      ok(null != ent);
    });

    test('stream', (t) async {
      // stream() runs the list op through the full pipeline and yields each
      // result item. Seed two entities via test mode; with the `streaming`
      // feature active it yields the feature's incremental items, else it
      // falls back to the materialised items — either way every item yields.
      final seed = <String, dynamic>{
        'entity': {
          'output_activate_portal_module': {
            'strm01': <String, dynamic>{'id': 'strm01'},
            'strm02': <String, dynamic>{'id': 'strm02'},
          }
        }
      };

      final sdkopts = <String, dynamic>{};
      if (null != config.feature['streaming']) {
        sdkopts['feature'] = {
          'streaming': {'active': true}
        };
      }

      final testsdk = BluefinTecsUserBackofficeSDK.test(seed, sdkopts);
      final ent = testsdk.OutputActivatePortalModule();

      final seen = [];
      await for (final item in ent.stream('list', <String, dynamic>{})) {
        seen.add(item);
      }
      equal(2, seen.length);

      // Fallback: with streaming inactive, stream() still yields both items
      // from the materialised result.
      final plainsdk = BluefinTecsUserBackofficeSDK.test(seed);
      final plainent = plainsdk.OutputActivatePortalModule();
      final seen2 = [];
      await for (final item in plainent.stream('list', <String, dynamic>{})) {
        seen2.add(item);
      }
      equal(2, seen2.length);
    });

    test('basic', (t) async {

      final live = 'TRUE' == Platform.environment['BLUEFIN_TECS_USER_BACKOFFICE_TEST_LIVE'];
      for (final op in ['create']) {
        if (maybeSkipControl(t, 'entityOp', 'output_activate_portal_module.' + op, live)) {
          return;
        }
      }

      final setup = basicSetup();
      // The basic flow consumes synthetic IDs and field values from the
      // fixture (entity TestData.json). Those don't exist on the live API.
      // Skip live runs unless the user provided a real ENTID env override.
      if (true == setup['syntheticOnly']) {
        t.skip('live entity test uses synthetic IDs from fixture — set BLUEFIN_TECS_USER_BACKOFFICE_TEST_OUTPUT_ACTIVATE_PORTAL_MODULE_ENTID JSON to run live');
        return;
      }
      final client = setup['client'];
      final struct = setup['struct'];

      final isempty = struct.isempty;
      final select = struct.select;


      // CREATE
      final output_activate_portal_module_ref01_ent = client.OutputActivatePortalModule();
      dynamic output_activate_portal_module_ref01_data = setup['data']['new']['output_activate_portal_module']['output_activate_portal_module_ref01'];

      output_activate_portal_module_ref01_data = await output_activate_portal_module_ref01_ent.create(output_activate_portal_module_ref01_data);
      ok(null != output_activate_portal_module_ref01_data);


    });
  });
}


Map<String, dynamic> basicSetup([dynamic extra]) {
  final options = <String, dynamic>{};

  final entityDataFile = resolveTestPath(
      '../.sdk/test/entity/output_activate_portal_module/OutputActivatePortalModuleTestData.json');

  final entityDataSource = File(entityDataFile).readAsStringSync();

  final entityData = jsonDecode(entityDataSource);

  options['entity'] = entityData['existing'];

  var client = BluefinTecsUserBackofficeSDK.test(options, extra);
  final struct = client.utility().struct;
  final merge = struct.merge;
  final transform = struct.transform;

  dynamic idmap = transform(
      <dynamic>['output_activate_portal_module01', 'output_activate_portal_module02', 'output_activate_portal_module03'],
      <String, dynamic>{
        '`\$PACK`': <dynamic>[
          '',
          <String, dynamic>{
            '`\$KEY`': '`\$COPY`',
            '`\$VAL`': <dynamic>['`\$FORMAT`', 'upper', '`\$COPY`'],
          }
        ]
      });

  // Detect whether the user provided a real ENTID JSON via env var. The
  // basic flow consumes synthetic IDs from the fixture file; without an
  // override those synthetic IDs reach the live API and 4xx. Surface this
  // to the test so it can skip rather than fail.
  final idmapEnvVal =
      Platform.environment['BLUEFIN_TECS_USER_BACKOFFICE_TEST_OUTPUT_ACTIVATE_PORTAL_MODULE_ENTID'];
  final idmapOverridden =
      null != idmapEnvVal && idmapEnvVal.trim().startsWith('{');

  final env = envOverride({
    'BLUEFIN_TECS_USER_BACKOFFICE_TEST_OUTPUT_ACTIVATE_PORTAL_MODULE_ENTID': idmap,
    'BLUEFIN_TECS_USER_BACKOFFICE_TEST_LIVE': 'FALSE',
    'BLUEFIN_TECS_USER_BACKOFFICE_TEST_EXPLAIN': 'FALSE',
    'BLUEFIN_TECS_USER_BACKOFFICE_APIKEY': 'NONE',
  });

  idmap = env['BLUEFIN_TECS_USER_BACKOFFICE_TEST_OUTPUT_ACTIVATE_PORTAL_MODULE_ENTID'];

  final live = 'TRUE' == env['BLUEFIN_TECS_USER_BACKOFFICE_TEST_LIVE'];

  if (live) {
    client = BluefinTecsUserBackofficeSDK(merge([
      <String, dynamic>{
        'apikey': env['BLUEFIN_TECS_USER_BACKOFFICE_APIKEY'],
      },
      extra
    ]));
  }

  final setup = <String, dynamic>{
    'idmap': idmap,
    'env': env,
    'options': options,
    'client': client,
    'struct': struct,
    'data': entityData,
    'explain': 'TRUE' == env['BLUEFIN_TECS_USER_BACKOFFICE_TEST_EXPLAIN'],
    'live': live,
    'syntheticOnly': live && !idmapOverridden,
    'now': DateTime.now().millisecondsSinceEpoch,
  };

  return setup;
}

